### ISSUE TITLE
Binary error does not change much, what can be a cause?

### ISSUE THREAD

**Issue by dprze** [vlfeat/matconvnet#917]
I am training quite small convolutional neural network for binary classification and I am stucked with problem, which I find hard to understand: while training, binary error stays at proximetly same value. What I have done trying to solve this problem:

1. Changing learning rate: generally I use three different learning rate values which I decrease during trainig. Changing those values had no effect.
2. I was experimenting with batch size, and very bizarre (for me, I am quite new in training CNNs, my previous experiments were with example models you could find on Internet): when batch size is smaller, binary error is smaller, for example: setting batch size to 5 led to binarry error ~2.8, while with batch size 10 binary error was ~5.8 (this dependence is monotonically increasing). Up to this day I thought that bigger batch size could help optimizing trainig, but has no effect on final error.

What is also important, train and validation error have similar (not high) values - from that I conclude, that overfitting is not a problem here. My database have total of 4828 examples with classes {-1, 1} and I am using 'binary' as error function.

Can a small database be a problem? Or should I maybe do far more epochs? I would very much appreciate some hints what can be a cause of that problem, as I've mentioned before I am just starting my adventure with CNNs and it's somehow confusing. Thanks a lot.
![binerr](https://cloud.githubusercontent.com/assets/26357206/23830970/53d89ea6-0717-11e7-9e96-e0db354253c1.jpg)
